core/src/avm2/globals/flash/system/WorkerDomain.as

```diff
@@ -1,10 +1,26 @@
 package flash.system {
+
+    import flash.utils.ByteArray;
+    import flash.system.Worker;
+    import __ruffle__.stub_method;
+
     [API("680")] // the docs say 682, that's wrong
     public final class WorkerDomain {
         public static const isSupported:Boolean = false;
+        
+        private static var _current:WorkerDomain;
+        
+        public static function get current():WorkerDomain {
+            if (!_current) {
+                _current = new WorkerDomain();
+            }
+            return _current;
+        }
+
+        public function createWorker(swf:ByteArray, giveAppPrivileges:Boolean = false):Worker {
+            stub_method("flash.system.WorkerDomain", "createWorker");
 
-        public function WorkerDomain() {
-            throw new ArgumentError("Error #2012: WorkerDomain$ class cannot be instantiated.", 2012)
```

Comment:
This constructor should be left as-is, the error should be mentioned in a comment and a stub warning should be produced